### PR TITLE
🌱 add runtime-extension to clusterctl default providers

### LINKED ISSUE #9796: Remove example from --runtime-extension flag in clusterctl
The `clusterctl` help text for the `--runtime-extension` options provides [an example](https://github.com/cahillsf/cluster-api/blob/aacbbc780a42363b28c232cccc87bfecd816df0b/cmd/clusterctl/cmd/init.go#L106C2-L106C2):
`Runtime extension providers and versions (e.g. test:v0.0.1) to add to the management cluster.`

Applying this option and example param results in the following error:
`Error: failed to get configuration for the RuntimeExtensionProvider with name test. Please check the provider name and/or add configuration for new providers using the .clusterctl config`

### PR DESCRIPTION
**Pull request by cahillsf**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #9796 (and bumps clusterctl help text versions for core capi components)

/area clusterctl

### CODE REVIEW

File: cmd/clusterctl/client/config/providers_client.go [hunk 1]
```diff
@@ -88,6 +88,11 @@ const (
 	RKE2ControlPlaneProviderName              = "rke2"
 )
 
+// Runtime extension providers.
+const (
+	RuntimeExtensionTestProviderName = "runtime-test"
```
[the remark this answers is not in the document]
Comment (cahillsf) [reply cmd/clusterctl/client/config/providers_client.go:93]: whatever you think is best 👍 -- will wait to see if Fabrizio weighs in prior to making further changes

[the remark this answers is not in the document]
Comment (sbueringer) [reply cmd/clusterctl/client/config/providers_client.go:93]: Sounds good. I'm also not sure what a good name is given this is user-facing (most of the places where we used "test-extension" before are just internal)

[the remark this answers is not in the document]
Comment (cahillsf) [reply cmd/clusterctl/client/config/providers_client.go:93]: 👋 @fabriziopandini bumping for when you have a chance to review

**Comment by fabriziopandini** [comment]
/hold
sorry for getting back to this late but I don't think we should expose test extensions without some docs about what they are and how to use them. Let's sort this out in the issue before moving on.

**Comment by cahillsf** [comment]
as discussed on the issue, keeping this component internal for the time being

**Pull request closed without merging**